src/watchers/file.ts

```diff
@@ -0,0 +1,63 @@
+import * as fs from "fs";
+import {Watcher} from "./watcher";
+import {ID} from "../runtime/runtime";
+
+class FileWatcher extends Watcher {
+
+    setup() {
+        let {program:me} = this;
+        
+        /*
+        fs.readFile("..\\foo.txt", {encoding: 'utf-8'}, function(err,data){
+            if (!err) {
+                console.log('received data: ' + data);
+            } else {
+                console.log(err);
+            }
+        });*/
+
+        me.watch("read a file", ({find, record}) => {
+            let readfile = find("file/read");
+            return [
+                record({readfile, path: readfile.path})
+            ]
+        })
+
+        me.asObjects<{readfile:ID, path:string}>(({adds, removes}) => {
+            Object.keys(adds).forEach((id) => {
+                let {readfile, path} = adds[id];
+                fs.readFile(path, {encoding: 'utf-8'}, function(err,data){
+                    if (!err) {
+                        let changes:any[] = [];
+                        changes.push(
+                            [readfile, "contents", data, 1],
+                        );
+                        me.inputEAVs(changes);
+                    } else {
+                        console.log(err);
```

Comment:
This should probably drop in an error message of some sort. We haven't fleshed out what the story for #error is yet, but it'd be perfectly fine to have `readfile.error += [code: <err code>, name:, text]` for the program to look for and handle.